eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/CommonHttpSinkHandler.java

```diff
@@ -0,0 +1,197 @@
+    }
+
+    /**
+     * Get the oldest data in the queue
+     *
+     * @return received data
+     */
+    public Object getReceivedData() {
+        if (!this.connectorConfig.isWebhook()) {
+            return null;
+        }
+        return this.receivedDataQueue.poll();
+    }
+
+    /**
+     * Get all received data
+     *
+     * @return all received data
+     */
+    public Object[] getAllReceivedData() {
+        if (!connectorConfig.isWebhook() || receivedDataQueue.isEmpty()) {
+            return new Object[0];
+        }
+        Object[] arr = receivedDataQueue.toArray();
+        receivedDataQueue.clear();
+        return arr;
+    }
+
+
+    @Override
+    public void start() {
+        // Create WebClient
+        doInitWebClient();
+    }
+
+    private void doInitWebClient() {
+        final Vertx vertx = Vertx.vertx();
+        WebClientOptions options = new WebClientOptions()
+            .setDefaultHost(this.connectorConfig.getHost())
+            .setDefaultPort(this.connectorConfig.getPort())
+            .setSsl(this.connectorConfig.isSsl())
+            .setKeepAlive(this.connectorConfig.isKeepAlive())
+            .setKeepAliveTimeout(this.connectorConfig.getKeepAliveTimeout() / 1000)
+            .setIdleTimeout(this.connectorConfig.getIdleTimeout())
+            .setIdleTimeoutUnit(TimeUnit.MILLISECONDS)
+            .setConnectTimeout(this.connectorConfig.getConnectionTimeout())
+            .setMaxPoolSize(this.connectorConfig.getMaxConnectionPoolSize());
+        this.webClient = WebClient.create(vertx, options);
+    }
+
+
+    @Override
+    public void handle(ConnectRecord record) {
+        // create headers
+        MultiMap headers = HttpHeaders.headers()
+            .set(HttpHeaderNames.ACCEPT, "application/json; charset=utf-8");
+
+        // convert ConnectRecord to HttpConnectRecord
+        HttpConnectRecord httpConnectRecord = convertToHttpConnectRecord(record);
+
+        // send the request
+        this.webClient.post(this.connectorConfig.getPath())
+            .putHeaders(headers)
+            .sendJson(httpConnectRecord)
+            .onSuccess(res -> {
+                Long timestamp = record.getTimestamp();
+                Map<String, ?> offset = record.getPosition().getOffset().getOffset();
+                log.info("Request sent successfully. Record: timestamp={}, offset={}", timestamp, offset);
+                // Determine whether the status code is 200
+                if (res.statusCode() == HttpResponseStatus.OK.code()) {
+                    // store the received data, when webhook is enabled
+                    if (this.connectorConfig.isWebhook()) {
+                        String dataStr = res.body().toString();
+                        if (dataStr.isEmpty()) {
+                            log.warn("Received data is empty.");
+                            return;
+                        }
+                        JSONObject receivedData = JSON.parseObject(dataStr);
```

Comment:
Serializing a response body and deserialize it again is a lot performance overhead. I think there's no need to output a warning when response body is empty.

By the way, the `bodyAsString()` method has a better performance than `toString()`.

Reply:
`bodyAsString()` operates an existing `Buffer` and can reduce the performance overhead of transforming String.

Reply:
> `bodyAsString()` operates an existing `Buffer` and can reduce the performance overhead of transforming String.

This doesn't convince me. `bodyAsBuffer()` simply adds an `instanceof` operation compared to `body()` method. Both methods do not involve any `new` operations, which seems to indicate that both are operating on existing `Buffer` objects. Is there still a difference in the lower-level part?

```java
@Override
public T body() {
  return body;
}

@Override
public Buffer bodyAsBuffer() {
  return body instanceof Buffer ? (Buffer) body : null;
}
```

Reply:
Indeed, their source code performance is similar. This is something that Vert.x does very well, and you've looked at it in detail. Considering that `bodyAsString()` has the `@CacheReturn` annotation, the overall performance might be slightly better.

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/fe2b7327b744fcf0f3ea3576c892c551ed1d6fb4